Batteries/Data/List/Basic.lean

```diff
@@ -736,7 +736,15 @@ def sigmaTR {σ : α → Type _} (l₁ : List α) (l₂ : ∀ a, List (σ a)) :
 ofFn f = [f 0, f 1, ... , f (n - 1)]
 ```
 -/
-def ofFn {n} (f : Fin n → α) : List α := (Array.ofFn f).data
+def ofFn {n} (f : Fin n → α) : List α := go n (Nat.le_refl _) where
+  /-- Auxillary for `List.ofFn`. -/
+  -- This used to be defined via `Array.ofFn`
+  -- but `Array.ofFn.go` is defined by well-founded recursion (slow to reduce)
+  -- and used `Array.push` (quadratic complexity on lists). Since mathlib relies on
```

Comment:
I don't know what is meant by this: `Array.push` is not quadratic complexity on arrays, and it's an array function.

Reply:
I assumes that the original definition, `List.ofFn = Array.ofFn` would in terms of definitional equality use the logical definition of the Array functions, which work on Arrays as if they are structures with a fields `.data` which is a list, and thus `Array.push` is actually `List.concat`. Am I mistaken here?